From 981f0d656d129a15bae5fc7c7c7f4c80f0a94c15 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 28 Aug 2023 18:26:43 +0800
Subject: [PATCH] [riscv] Reduce the vector arch code (Part 8)

Delete unused arch code.

Change-Id: I45b5dfed1e45c0134a9e48f7618a8fcd929e2da9
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4816308
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#89648}
---
 .../backend/riscv/instruction-codes-riscv.h     | 17 -----------------
 .../riscv/instruction-scheduler-riscv.cc        | 17 -----------------
 2 files changed, 34 deletions(-)

diff --git a/src/compiler/backend/riscv/instruction-codes-riscv.h b/src/compiler/backend/riscv/instruction-codes-riscv.h
index 822d5687c5c..72d92878794 100644
--- a/src/compiler/backend/riscv/instruction-codes-riscv.h
+++ b/src/compiler/backend/riscv/instruction-codes-riscv.h
@@ -283,13 +283,6 @@ namespace compiler {
   V(RiscvS128Load64Zero)                  \
   V(RiscvS128Load32Zero)                  \
   V(RiscvV128AnyTrue)                     \
-  V(RiscvS32x4InterleaveRight)            \
-  V(RiscvS32x4InterleaveLeft)             \
-  V(RiscvS32x4PackEven)                   \
-  V(RiscvS32x4PackOdd)                    \
-  V(RiscvS32x4InterleaveEven)             \
-  V(RiscvS32x4InterleaveOdd)              \
-  V(RiscvS32x4Shuffle)                    \
   V(RiscvS16x8InterleaveRight)            \
   V(RiscvS16x8InterleaveLeft)             \
   V(RiscvS16x8PackEven)                   \
@@ -298,17 +291,7 @@ namespace compiler {
   V(RiscvS16x8InterleaveOdd)              \
   V(RiscvS16x4Reverse)                    \
   V(RiscvS16x2Reverse)                    \
-  V(RiscvS8x16InterleaveRight)            \
-  V(RiscvS8x16InterleaveLeft)             \
-  V(RiscvS8x16PackEven)                   \
-  V(RiscvS8x16PackOdd)                    \
-  V(RiscvS8x16InterleaveEven)             \
-  V(RiscvS8x16InterleaveOdd)              \
   V(RiscvI8x16Shuffle)                    \
-  V(RiscvS8x16Concat)                     \
-  V(RiscvS8x8Reverse)                     \
-  V(RiscvS8x4Reverse)                     \
-  V(RiscvS8x2Reverse)                     \
   V(RiscvS128LoadSplat)                   \
   V(RiscvS128Load64ExtendS)               \
   V(RiscvS128Load64ExtendU)               \
diff --git a/src/compiler/backend/riscv/instruction-scheduler-riscv.cc b/src/compiler/backend/riscv/instruction-scheduler-riscv.cc
index cae191a6b98..b8778958798 100644
--- a/src/compiler/backend/riscv/instruction-scheduler-riscv.cc
+++ b/src/compiler/backend/riscv/instruction-scheduler-riscv.cc
@@ -217,23 +217,6 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kRiscvS16x2Reverse:
     case kRiscvS16x4Reverse:
     case kRiscvV128AnyTrue:
-    case kRiscvS32x4InterleaveEven:
-    case kRiscvS32x4InterleaveOdd:
-    case kRiscvS32x4InterleaveLeft:
-    case kRiscvS32x4InterleaveRight:
-    case kRiscvS32x4PackEven:
-    case kRiscvS32x4PackOdd:
-    case kRiscvS32x4Shuffle:
-    case kRiscvS8x16Concat:
-    case kRiscvS8x16InterleaveEven:
-    case kRiscvS8x16InterleaveOdd:
-    case kRiscvS8x16InterleaveLeft:
-    case kRiscvS8x16InterleaveRight:
-    case kRiscvS8x16PackEven:
-    case kRiscvS8x16PackOdd:
-    case kRiscvS8x2Reverse:
-    case kRiscvS8x4Reverse:
-    case kRiscvS8x8Reverse:
     case kRiscvI8x16Shuffle:
     case kRiscvVwmul:
     case kRiscvVwmulu:
-- 
2.35.1

